modules/social_features/social_group/social_group.module

```diff
@@ -2205,44 +2205,33 @@ function social_group_user_cancel_methods_alter(&$methods) {
 /**
  * Implements hook_social_user_account_header_create_links().
  *
- * Adds the "Create Group" link to the content creation menu.
- * When the user can only create groups of one type, the user skips the
- * group type selection page.
+ * Adds the "Create Group" link to the content creation menu, that redirect user
+ * to the create form of flexible group. If the user doesn't have access the
+ * link won't be available.
  */
 function social_group_social_user_account_header_create_links($context) {
-
-  // If we have a user to work with, we can check if we should redirect it to
-  // the create group overview or a single group.
-  // If we do not have a user we'll default to the overview.
-  $route_add_group = NULL;
   if (!empty($context['user'])) {
     /** @var \Drupal\Core\Session\AccountInterface $user */
     $account = $context['user'];
     /** @var \Drupal\Core\Url|NULL $route_add_group */
     $route_add_group = \Drupal::service('social_group.helper_service')
       ->getGroupsToAddUrl($account);
 
```

Comment:
This contradicts the proposed resolution.
> Update social_group_social_user_account_header_create_links to point directly to the flexible group add form

There shouldn't be a need to use this service anymore (though we may need to move this hook to `social_group_flexible_group`) since we know we can use the group entity create route with the `flexible_group` bundle. See `social_topic_social_user_account_header_create_links` for an example.

Reply:
done